lib/signs_ui/alerts/events.ex

```diff
@@ -0,0 +1,48 @@
+defmodule SignsUi.Alerts.Events do
+  alias SignsUi.Alerts.Alert
+  alias SignsUi.Alerts.State
+
+  @spec parse(String.t() | list() | map()) :: [State.row()] | State.row()
+  def parse(payload) when is_binary(payload) do
+    {:ok, decoded} = Jason.decode(payload)
+
+    parse(decoded)
+  end
+
+  def parse(payload) when is_list(payload) do
+    Enum.map(payload, &parse/1)
+  end
+
+  def parse(payload) when is_map(payload) do
+    id = payload["id"]
+
+    {created_at, service_effect, routes} = parse_attributes(payload["attributes"])
+
+    {id,
+     %{
+       created_at: created_at,
+       service_effect: service_effect,
+       affected_routes: routes
+     }}
+  end
+
+  @spec parse_attributes(nil | map()) ::
+          {DateTime.t() | nil, String.t() | nil, MapSet.t(Alert.route_id()) | nil}
+  defp parse_attributes(nil), do: {nil, nil, nil}
+
+  defp parse_attributes(attributes) do
```

Comment:
Looking at the definition of `Alert.t()`, we allow `created_at` and `service_effect` to both be `nil`, so if that's the case for either of them, we should still be able to get a valid `%Alert{}` out of the data.

Reply:
I'm not sure what case you're referencing here but if it's `parse_attributes(nil)`, that case only matches if the actual `attributes` field in the event is nil, which means we can't get either `created_at` or `service_effect` out of it.